### DISCUSSION TITLE
Font color of properties in inspector

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2618] [Q&A]
### Introduction

I have a property called 'Location Name' in inspector as shown below. I wanted to mark it as a mandatory field by keeping an asterisk symbol at the beginning of the label. Is it possible to make the asterisk red in color and leave the rest of the text as it is?

```
text: {
type: 'content-editable',
label: '*Location Name',
group: 'text',
index: 1,
}
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
You can define the label as an HTML string:
```ts
text: {
  type: 'content-editable',
  label: '<span style="color:red">*</span>Text',
  group: 'text',
  index: 1,
}
```
Or you can define a custom label using the [renderLabel()](https://resources.jointjs.com/docs/rappid/v4.0/ui.Inspector.html#ui.Inspector.custom-labels) option.

**Reply by kvinita** [reply]
I used to first approach that you suggested using a span tag
`text: {
                type: 'number',
                label: '<span style="color:red">*</span>Text',
                group: 'text',
                index: 3,
              }`
But it was rendered like shown below:
![image](https://github.com/clientIO/joint/assets/124018900/4599a060-55a9-4258-9ef2-50583ad8cf3c)

So, next I tried to create Custom Label as below:
         `renderLabel: function (opt: any, path: any) {
          var $label = document.createElement('label');
          var $el = $label;

          if (opt.label && opt.label.startsWith('*')) {
            var $span = document.createElement('span');
            $span.textContent = '*';
            $span.style.color = 'red';
            $label.appendChild($span);
            var labelText = opt.label.substring(1);
            $label.appendChild(document.createTextNode(labelText));
          } else {
            $label.textContent = opt.label || path;
          }

          $label.style.cssText = 'display:block !important;';

          if (opt.url) {
            var $a = document.createElement('a');
            $a.href = opt.url;
            $a.target = 'blank';
            $a.appendChild($el);
            $el = $a;
          }
          return $el;
        }`

This raised an error at `$el = $a;` line. It says `"Type 'HTMLAnchorElement' is missing the following properties from type 'HTMLLabelElement': control, form, htmlFor"`.

FYI, I'm using Angular16 as the front end framework.
Can you please let me know what the issue is with the code?
Any leads would be appreciated.

**Reply by kumilingus** [reply]
If it renders like this:
![image](https://github.com/clientIO/joint/assets/124018900/4599a060-55a9-4258-9ef2-50583ad8cf3c)
You are most likely using an older version of *JointJS+*.

---
`"Type 'HTMLAnchorElement' is missing the following properties from type 'HTMLLabelElement': control, form, htmlFor"`.
It is a general TypeScript error. You can assign type `HTMLLabelElement` to  `HTMLAnchorElement`.

For example, try creating 2 variables or make `$el` type more generic.
